_posts/2019-02-06-Age-Targeting-Part2.md

```diff
@@ -0,0 +1,237 @@
+<p align="center"> <img src="../images/Age-Targeting/weight_class.png"> </p>
+As you can see the best features of each class are the apps used.
+Moreover, this list make extremely sense with what you would expect.
+PUT SOMETHING HERE WHICH DESCRIBE THE APPS IN THE LIST
+This gives additional confidence on putting the model into production.
+
+Analyzing the output where the model fails is often useful.
+In the plot you can see the distribution of the difference between the real age and the estimated age_group.
+Keep in mind that every class span about 10 years and so a value of the showed difference aroung -10, 10 correspond with a deviceid whose age_group is missclassified with a adjacent class.
+For example, if the age of the device is 36 and the model predict the class 25_34 we have a distance of 36 -34 = 2 ∈ [-10, 10].
+Big mistakes, such as missidentify a younger with the older and vice versa, occurs rarely in comparison to the adjacent-like mistakes. 
+<p align="center"> <img src="../images/Age-Targeting/diff_age_agegroup.png"> </p>
+When we analyze those devices we see that their app usage would be commonly associated to the other classes.
+In the end you know, it is not important the age you have but the one you feel.
+
+
+**Predict over the entire dataset**<br/>
+The model gives as output the five probabilities for a device to belong to one of the age_group categories.
+Comparing the output distribution for every probabilities between the ground truth and the rest of the devices, we can see that there are some differences.
+In the picture you have an example. THE PICTURE NEED TO BE CHANGED, EVEN A REWIGHTIED HISTOGRAM COULD HELP THE COMPREENSION HERE.
+<p align="center"> <img src="../images/Age-Targeting/output_predict_dist_ground_truth.png"> </p>
+<p align="center"> <img src="../images/Age-Targeting/output_predict_dist_overall.png"> </p>
+These differences are not synonimous of overfitting (that can be excluded because in the test set the distributions are very similar).
+In fact, the ground truth is a special part of the entire dataset because it contains the devices which visit the apps that provide the age information.
+With other words, the ground truth has a bias wrt to the final dataset where the model will be applied.
+If we would estimate any figure of merit from the ground truth as it is we would end up with a bias, non realistic measure.
+To fix the bias we reweight the ground truth distribution using the overall shape.
+We can say that, for a fixed bin in the distribution, we take the number of devices from the overall one, while the purity for the bin is estimated from the trained dataset.
+The reweight is performed separatelly for each country because the output is country-dependent.
+
+After the reweight and independently for every groups we calculate the thresold the corresponding probability should have so that a device will belong to that class with a required precision. 
```

Comment:
After the reweighting is done independently for each group, we calculate a threshold. The threshold corresponds to a certain precision and is determined for each class and country. Therefore, we can predict the age group of all devices for each country and classes with the same confidence.